### PR TITLE
Restore the superbuild

### PR DESCRIPTION
**Pull request by benson31**
It is with great sadness that this is done. The superbuild has been kept for internal use and for those situations in which real package managers prove inadequate, which were supposed to be few. Yet I have had to hand it out to nearly every user and/or developer of LBANN, because I guess our simple little package is too much for certain package managers to handle. Thus, here it is.

Do note that I do not want to spend any more of my life maintaining build systems or hand-holding developers (or users) than is programmatically necessary. Therefore this is presented as-is, with a couple of example scripts and a README. Good luck.

Finally, do continue to relentlessly hound the developers and/or maintainers of whichever package manager has failed to provide you with a working LBANN build so that this superbuild can fade away for a final time. It's not a hard package to build, and the dependencies are both quite few and quite common. So it's a real shame that there is so much trouble.

### CODE REVIEW

**Review by ndryden — APPROVED** [review]
🥲

File: scripts/superbuild/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,180 @@
+################################################################################
+## Copyright (c) 2014-2022, Lawrence Livermore National Security, LLC.
```
Comment (tbennun) [inline scripts/superbuild/CMakeLists.txt:2]: Copyright year

File: scripts/superbuild/adiak/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,72 @@
+################################################################################
+## Copyright (c) 2014-2022, Lawrence Livermore National Security, LLC.
```
Comment (tbennun) [inline scripts/superbuild/adiak/CMakeLists.txt:2]: Year

File: scripts/superbuild/aluminum/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,67 @@
+################################################################################
+## Copyright (c) 2014-2022, Lawrence Livermore National Security, LLC.
```
Comment (tbennun) [inline scripts/superbuild/aluminum/CMakeLists.txt:2]:
```suggestion
## Copyright (c) 2014-2024, Lawrence Livermore National Security, LLC.
```

**Review by tbennun — APPROVED** [review]

**Pull request merged**